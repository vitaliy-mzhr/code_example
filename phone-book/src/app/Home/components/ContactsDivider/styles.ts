import styled from 'styled-components';

const ContactsDividerStyles = styled.div`
  .divider {
    padding: .5rem;
    margin-bottom: .25rem;
    background-color: ${props => props.theme.colors.greyColor};
  }
  .title {
    text-align: center;
  }
`;

export default ContactsDividerStyles;
