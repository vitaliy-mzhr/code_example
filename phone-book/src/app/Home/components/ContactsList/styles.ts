import styled from 'styled-components';

const ContactsListStyles = styled.div`
  width: 100%;
  max-width: ${props => props.theme.dimensions.contactsMaxWidthRem * 1.25}rem;
  margin: 2rem auto;
  
  @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {
    max-width: ${props => props.theme.dimensions.contactsMaxWidthRem}rem;
  }
`;

export default ContactsListStyles;
