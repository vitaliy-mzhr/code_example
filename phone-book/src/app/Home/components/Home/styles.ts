import styled from 'styled-components';

const HomeStyles = styled.div`
  height: 100%;
  @media all and (max-width: ${props => props.theme.media.mobileMaxWidthPx}px) {

  }
`;

export default HomeStyles;
