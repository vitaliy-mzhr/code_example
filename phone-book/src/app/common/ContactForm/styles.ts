import styled from 'styled-components';

const ContactFormStyles = styled.div`
  h3 {
    text-align: center;
  }
  .form-container {
    margin: 0 auto;
    padding-top: 2rem;
    max-width: 20rem;
    display: flex;
    flex-direction: column;
    justify-content: center;
  }
  .buttons-container {
    display: flex;
    justify-content: center;
    padding: 1rem;
    button {
      margin: 0 .25rem;
    }
  }
`;

export default ContactFormStyles;
