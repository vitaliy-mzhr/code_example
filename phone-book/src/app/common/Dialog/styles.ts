import styled from 'styled-components';

const AlertDialogStyles = styled.div`
  min-width: 14rem;
  .dialog-title {
    h2 {
      text-align: center;
    }
  }
  .buttons-container {
    display: flex;
    justify-content: space-evenly;
    align-items: center;
  }
  
  .confirm-button-icon {
    color: ${props => props.theme.colors.greenColor};
  }
  .cancel-button-icon {
    color: ${props => props.theme.colors.errorColor};
  }
`;

export default AlertDialogStyles;
