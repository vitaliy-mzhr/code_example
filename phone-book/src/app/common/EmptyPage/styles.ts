import styled from 'styled-components';

const EmptyPageStyles = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  .page-message-container {
    margin-top: 2rem;
    padding: 2rem;
}
`;

export default EmptyPageStyles;
