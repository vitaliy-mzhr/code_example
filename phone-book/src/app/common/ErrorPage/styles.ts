import styled from 'styled-components';

const ErrorPageStyles = styled.div`
  display: flex;
  justify-content: center;
  width: 100%;
  .page-message-container {
    margin-top: 2rem;
    padding: 2rem;
    color: ${props => props.theme.colors.errorColor};
}
`;

export default ErrorPageStyles;
