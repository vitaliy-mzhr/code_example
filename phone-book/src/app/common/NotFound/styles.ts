import styled from 'styled-components';

const NotFoundStyles = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  
  a {
    text-decoration: none;
  }
`;

export default NotFoundStyles;
