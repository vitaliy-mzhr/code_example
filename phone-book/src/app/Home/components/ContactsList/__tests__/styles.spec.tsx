import * as React from 'react';
import ContactsListStyles from '../styles';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import { mainTheme } from '../../../../../styles/themes';

it('ContactsListStyles applied correctly', () => {
  expect(shallow(<ContactsListStyles theme={mainTheme} />)).toMatchSnapshot();
});