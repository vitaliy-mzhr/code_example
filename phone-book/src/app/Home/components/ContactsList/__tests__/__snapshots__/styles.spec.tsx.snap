// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactsListStyles applied correctly 1`] = `
.c0 {
  width: 100%;
  max-width: 26.25rem;
  margin: 2rem auto;
}

@media all and (max-width:768px) {
  .c0 {
    max-width: 21rem;
  }
}

<div
  className="c0"
/>
`;
